custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023-2024 Broadcom. All rights reserved.
+
+package dev.gemfire.stat.logger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.geode.StatisticDescriptor;
+import org.apache.geode.Statistics;
+import org.apache.geode.StatisticsType;
+import org.apache.geode.cache.*;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionContext;
+import org.apache.geode.distributed.DistributedSystem;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class StatLogger implements Function,Declarable {
+  private static final Logger logger = LogService.getLogger();
+  private List<StatsHolder> listOfStats = new CopyOnWriteArrayList<>();
+  private Timer timer;
```

Comment:
I'm confused why this class has instance variables but also stores the Timer instance in a local Region. I think all these instance variables can instead of local variables created in execute and passed around as parameters to the other methods.

Reply:
@BalaKaza as per our discussion yesterday. I would still prefer that we keep as little state as possible in the Function. I don't think we need to keep ANY of the stateful fields in the function at all... The list of stats are only required until timer is created. 

In order to always make sure that we get the correct timer, we need to get it from the LOCAL region... etc...

I think @dschneider-pivotal is correct. We should remove as many of the instance variable fields as possible.

Reply:
Yup @kohlmu-pivotal I cleaned it up a bit now. Let me know if you find anything weird.